### PR TITLE
cluster: remove salt

### PR DESCRIPTION
**Pull request by mikedanese**
Nothing uses it directly. There were some dangling dependencies from GCE and Centos kube-ups, so I just copied them into respective dirs.

see #49213

```release-note
Remove deprecated and unmaintained salt support. kubernetes-salt.tar.gz will no longer be published in the release tarball.
```

### CODE REVIEW

File: build/lib/release.sh [hunk 1]
```diff
@@ -391,38 +391,35 @@ function kube::release::package_salt_tarball() {
...
   kube::log::status "Building tarball: manifests"
 
-  local salt_dir="${KUBE_ROOT}/cluster/saltbase/salt"
+  local src_dir="${KUBE_ROOT}/cluster/gce/manifests"
 
   local release_stage="${RELEASE_STAGE}/manifests/kubernetes"
   rm -rf "${release_stage}"
 
-  mkdir -p "${release_stage}"
-  cp "${salt_dir}/kube-registry-proxy/kube-registry-proxy.yaml" "${release_stage}/"
```
Comment (roberthbailey) [inline build/lib/release.sh]: These files moved to a different dir in the tar, but I don't see a corresponding change in the node setup scripts.

Comment (mikedanese) [reply build/lib/release.sh]: changes are in `cluster/gce/gci/configure-helper.sh`

Comment (roberthbailey) [reply build/lib/release.sh]: I see it now. thanks.

**Comment by mikedanese** [comment]
@roberthbailey all green

**Comment by roberthbailey** [comment]
woo!

**Pull request merged**